src/test/java/org/redcrosswarriors/controller/TestRequestBloodController.java

```diff
@@ -0,0 +1,62 @@
+package org.redcrosswarriors.controller;
+
+import com.google.gson.Gson;
+import org.junit.jupiter.api.Test;
+import org.redcrosswarriors.controllerservice.ProfileControllerService;
+import org.redcrosswarriors.controllerservice.RegistrationControllerService;
+import org.redcrosswarriors.controllerservice.RequestBloodControllerService;
+import org.redcrosswarriors.model.input.RegistrationInput;
+import org.redcrosswarriors.model.input.RequestBloodInput;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class TestRequestBloodController {
+
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    RequestBloodControllerService service;
+
+    @Test
+    public void testRequestBlood() throws Exception
```

Comment:
test fails. You need to add @WithMockUser(username="admin@wayne.edu",roles={"USER","ADMIN"}) before the test so that it attempts to get the information as a logged in user